centreon-dsm/www/modules/centreon-dsm/core/configuration/services/DB-Func.php

```diff
@@ -153,24 +183,32 @@ function disablePoolInDB($pool_id = null, $pool_arr = array())
     }
 
     foreach ($pool_arr as $id => $values) {
-        $pearDB->query("UPDATE mod_dsm_pool SET pool_activate = '0' WHERE pool_id = '" . $id . "'");
+        if (!is_int($id)) {
```

Comment:
you can cast $Id into int before the test, and your test can be : if($id <= 0) it's safer because $i can be a string maybe :)